thrust/thrust/detail/function.h

```diff
@@ -31,119 +31,19 @@ THRUST_NAMESPACE_BEGIN
-  {
-    return static_cast<Result>(m_f(thrust::raw_reference_cast(x), thrust::raw_reference_cast(y)));
-  }
-
-  _CCCL_EXEC_CHECK_DISABLE
-  template <typename Argument1, typename Argument2>
-  _CCCL_FORCEINLINE _CCCL_HOST_DEVICE Result operator()(const Argument1& x, Argument2& y) const
-  {
-    return static_cast<Result>(m_f(thrust::raw_reference_cast(x), thrust::raw_reference_cast(y)));
-  }
-
-  _CCCL_EXEC_CHECK_DISABLE
-  template <typename Argument1, typename Argument2>
-  _CCCL_FORCEINLINE _CCCL_HOST_DEVICE Result operator()(const Argument1& x, const Argument2& y) const
-  {
-    return static_cast<Result>(m_f(thrust::raw_reference_cast(x), thrust::raw_reference_cast(y)));
-  }
-
-  _CCCL_EXEC_CHECK_DISABLE
-  template <typename Argument1, typename Argument2>
-  _CCCL_FORCEINLINE _CCCL_HOST_DEVICE Result operator()(Argument1& x, const Argument2& y) const
-  {
-    return static_cast<Result>(m_f(thrust::raw_reference_cast(x), thrust::raw_reference_cast(y)));
-  }
-}; // end wrapped_function
-
-// Specialize for void return types:
-template <typename Function>
-struct wrapped_function<Function, void>
-{
-  // mutable because Function::operator() might be const
-  mutable Function m_f;
-  inline _CCCL_HOST_DEVICE wrapped_function()
-      : m_f()
-  {}
-
-  inline _CCCL_HOST_DEVICE wrapped_function(const Function& f)
-      : m_f(f)
-  {}
-
-  _CCCL_EXEC_CHECK_DISABLE
-  template <typename Argument>
-  _CCCL_FORCEINLINE _CCCL_HOST_DEVICE void operator()(Argument& x) const
-  {
-    m_f(thrust::raw_reference_cast(x));
-  }
-
-  _CCCL_EXEC_CHECK_DISABLE
-  template <typename Argument>
-  _CCCL_FORCEINLINE _CCCL_HOST_DEVICE void operator()(const Argument& x) const
-  {
-    m_f(thrust::raw_reference_cast(x));
-  }
-
   _CCCL_EXEC_CHECK_DISABLE
-  template <typename Argument1, typename Argument2>
-  _CCCL_FORCEINLINE _CCCL_HOST_DEVICE void operator()(Argument1& x, Argument2& y) const
+  template <typename... Ts>
+  _CCCL_FORCEINLINE _CCCL_HOST_DEVICE Result operator()(Ts&&... args) const
   {
-    m_f(thrust::raw_reference_cast(x), thrust::raw_reference_cast(y));
-  }
-
-  _CCCL_EXEC_CHECK_DISABLE
-  template <typename Argument1, typename Argument2>
-  _CCCL_FORCEINLINE _CCCL_HOST_DEVICE void operator()(const Argument1& x, Argument2& y) const
-  {
-    m_f(thrust::raw_reference_cast(x), thrust::raw_reference_cast(y));
-  }
-  _CCCL_EXEC_CHECK_DISABLE
-  template <typename Argument1, typename Argument2>
-  _CCCL_FORCEINLINE _CCCL_HOST_DEVICE void operator()(const Argument1& x, const Argument2& y) const
-  {
-    m_f(thrust::raw_reference_cast(x), thrust::raw_reference_cast(y));
-  }
-  _CCCL_EXEC_CHECK_DISABLE
-  template <typename Argument1, typename Argument2>
-  _CCCL_FORCEINLINE _CCCL_HOST_DEVICE void operator()(Argument1& x, const Argument2& y) const
-  {
-    m_f(thrust::raw_reference_cast(x), thrust::raw_reference_cast(y));
+    return static_cast<Result>(m_f(thrust::raw_reference_cast(std::forward<Ts>(args))...));
```

Comment:
I believe this should be 
```suggestion
    return static_cast<Result>(m_f(thrust::raw_reference_cast(::cuda::std::forward<Ts>(args))...));
```

Reply:
I believe so as well. Thx!